humility-core/src/arch/rv.rs

```diff
@@ -85,10 +85,21 @@ impl Arch for RVArch {
     ) -> Result<u64> {
         match hubris.lookup_symword("CURRENT_TASK_PTR") {
             Ok(ptr) => Ok(core.read_word_32(ptr)? as u64),
-            // Means current task is in mscratch or sscratch, but only if on riscv
+            // Means current task is in mscratch or sscratch
             Err(_) => {
-                // TODO right now blindly check mscratch, but should check that it is valid
-                core.read_reg(Register::RiscV(RVRegister::MSCRATCH))
+                let task_register = if hubris
+                    .manifest
+                    .features
+                    .contains(&"s-mode".to_owned())
```

Comment:
Looking at the app.toml contained in the archive for 'demo-rv64-qemu-virt-supervisor' there do not appear to be any features set for '[kernel]' (humility populates manifest.features from the kernel features). Is there another change to ensure that 's-mode' is set as a kernel feature?

Reply:
Sorry, @tdewey-rivos has just informed me about https://github.com/rivosinc/hubris/pull/80

Reply:
This is, I believe, a particularly neat solution to determining the execution mode.